plugin-examples/src/main/java/com/hotels/styx/ExampleMessageIntercept.java

```diff
@@ -0,0 +1,78 @@
+package com.hotels.styx;
+
+import com.hotels.styx.api.Eventual;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.LiveHttpRequest;
+import com.hotels.styx.api.LiveHttpResponse;
+import com.hotels.styx.api.plugins.spi.Plugin;
+
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * This shows an example of an inteceptor that responds to a received request
+ */
+
+public class Example implements Plugin {
+
+    @Override
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
+        if (request.header("X-Respond").isPresent()) {
+            // request.consume();
+            return Eventual.of(HttpResponse.response(OK)
+                    .header(CONTENT_TYPE, "text/plain; charset=utf-8")
+                    .body("Responding from plugin", UTF_8)
+                    .build()
+                    .stream());
+        } else {
+            return chain.proceed(request);
+        }
+    }
+}
+
+/**
+ * You can replace live content
+ * If you need to replace a message body based on information in the message headers and don't care what is in
+ * the message body. (E.g you need to add a HTTP message body based on a HTTP error code)
+ *
+ * You can transform a live HTTP message body using a replaceWith Bytestream operator such as in the example.
+ */
+
+//Example:
+
+chain.proceed(request)
```

Comment:
We have dangling Java code here that is outside of any valid part of the file. This will prevent it from compiling. Presumably it is meant to be commented too.

However, it doesn't really make sense to have all these comments/code examples underneath the class file. If we have one example as a proper class with accompanying test, shouldn't the rest of the examples be done in the same way?